/* Copyright 2023 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package fedlearner_webconsole.proto.rpc.v2;

import "fedlearner_webconsole/proto/common.proto";
import "google/protobuf/struct.proto";


message CheckHealthRequest {}

message CheckHealthResponse {
  ApplicationVersion application_version = 1;
  bool healthy = 2;
  string message = 3;
}

message ListFlagsRequest {}

message ListFlagsResponse {
  google.protobuf.Struct flags = 1;
}

message CheckTeeEnabledRequest {}

message CheckTeeEnabledResponse {
  bool tee_enabled = 1;
}

service SystemService {
  rpc CheckHealth (CheckHealthRequest) returns (CheckHealthResponse) {}
  rpc ListFlags (ListFlagsRequest) returns (ListFlagsResponse) {}
  rpc CheckTeeEnabled (CheckTeeEnabledRequest) returns (CheckTeeEnabledResponse) {}
}
